Sources/EpoxyLayoutGroups/Types/SpringAnimationParameters.swift

```diff
@@ -0,0 +1,25 @@
+// Created by lindsay_pond on 5/13/22.
+// Copyright © 2022 Airbnb Inc. All rights reserved.
+
+import UIKit
+
+/// Provides parameter values, which can be passed to customize the animation on an Group's animated update (`_setItems(::)`).
+public struct SpringAnimationParameters: Hashable {
```

Comment:
Now that we're making this a public API, I think I would prefer that we write this in a slightly different way, e.g.:
```swift
public struct LayoutGroupUpdateAnimation: Hashable {
  public init(…) { … }
  public var id: AnyHashable
  public var animate: (_ animations: () -> Void, _ completion: (_ completed: Bool) -> Void)
}
```
Then we could define spring animations something like:
```swift
extension LayoutGroupUpdateAnimation {
  public static func spring(
    duration: TimeInterval = 0.5, 
    delay: TimeInterval = 0.0,
    dampingRatio: CGFloat = 1.0, 
    initialSpringVelocity: CGFloat = 0.0)
  {
    .init(id: AnimationID.spring) { animations, completion in
      UIView.animate(
          withDuration: duration,
          delay: delay,
          usingSpringWithDamping: dampingRatio,
          initialSpringVelocity: initialSpringVelocity,
          options: [.beginFromCurrentState, .allowUserInteraction],
          animations: animations
          completion: completion)
    }
  }
}

private enum AnimationID {
  case spring
}
```
Which enables full flexibility, e.g. if someone wants a non-spring animation

Reply:
@erichoracek -- recommendations on hashability? Is there a way to ignore the hashability on the closure? Or to ignore the LayoutGroupUpdateAnimation on the style?

Reply:
@lepond That's what I was imagining that the `id` would serve as, so the `Hashable` implementation would look something like:
```swift
extension LayoutGroupUpdateAnimation {
  public func hash(into hasher: inout Hasher) {
    hasher.combine(id)
  }

  public static func == (lhs: LayoutGroupUpdateAnimation, rhs: LayoutGroupUpdateAnimation) -> Bool {
    lhs.id == rhs.id
  }
}
```